test/fakeintake/server/parser.go

```diff
@@ -0,0 +1,77 @@
+package server
+
+import (
+	"encoding/json"
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/fakeintake/api"
+)
+
+type PayloadParser struct {
+	payloadJsonStore map[string][]api.ParsedPayload
+}
+
+func NewPayloadParser() PayloadParser {
+	parser := PayloadParser{
+		payloadJsonStore: map[string][]api.ParsedPayload{},
+	}
+	return parser
+}
+
+func (fi *PayloadParser) getLogPayLoadData(payload api.Payload) string {
```

Comment:
I think the naming can better reflect what the function does by adding JSON to the name of the function.
The same comment applies also for `getMetricPayLoadData` and `getCheckRunPayLoadData`